pkg/client/devops/jenkins/credential.go

```diff
@@ -264,15 +264,15 @@ func (j *Jenkins) UpdateCredentialInProject(projectId string, credential *v1.Sec
 		klog.Errorf("%+v", err)
 		return "", restful.NewError(http.StatusBadRequest, err.Error())
 	}
-	response, err := j.Requester.Post(
+	response, err := j.Requester.PostForm(
 		fmt.Sprintf("/job/%s/credentials/store/folder/domain/_/credential/%s/updateSubmit", projectId, credential.Name),
 		nil, nil, map[string]string{
 			"json": requestContent,
 		})
 	if err != nil {
 		return "", err
 	}
-	if response.StatusCode != http.StatusOK {
+	if response.StatusCode != http.StatusOK && response.StatusCode != http.StatusFound {
```

Comment:
Same question about the status code checking.

Reply:
The response to the request submitted to jenkins is `http.StatusFound` under normal circumstances.
If you don’t modify it in this way, there will be an error in the credential synced to Jenkins from ks

Reply:
Other contributors cannot understand it without the comment. So, please add it. It would be perfect if you can find the corresponding code lines of Jenkins.

Reply:
OK.I try to read the code of jenkins

Reply:
This [reference](https://github.com/jenkinsci/credentials-plugin/blob/72db2dbd65c877c1bc32c11064a139cc4bb71325/src/main/java/com/cloudbees/plugins/credentials/CredentialsStoreAction.java#L799) might be helpful for you.